Runtime/Utils/Values/SubscribableValue.cs

```diff
@@ -7,29 +7,52 @@ public class SubscribableValue<T>
 	{
 		private T value;
 
+		[field: NonSerialized]
 		public event Action<T> OnValueChanged;
 
 		public T Value
 		{
-			get
-			{
-				return value;
-			}
+			get => value;
 			set
 			{
-				this.value = value;
-
-				// The '?' operator doesn't work with the serialization
-				if (OnValueChanged != null)
+				if (Equals(this.value, value))
 				{
-					OnValueChanged.Invoke(this.value);
+					return;
 				}
+				
+				this.value = value;
```

Comment:
Extract these two lines into a method so we don't repeat logic in the ForceValueChange